### PR TITLE
5118 Implement MathJax JS hook to update contents after live update

### PR DESCRIPTION
**Pull request by abanink**
This uses the JS hook mechanism from PR https://github.com/friendica/friendica/issues/5179 to solve issue https://github.com/friendica/friendica/issues/5118

### CODE REVIEW

File: mathjax/mathjax.php [hunk 1]
```diff
@@ -16,12 +16,19 @@ function mathjax_install() {
     Addon::registerHook('page_header', 'addon/mathjax/mathjax.php', 'mathjax_page_header');
     Addon::registerHook('addon_settings', 'addon/mathjax/mathjax.php', 'mathjax_settings');
     Addon::registerHook('addon_settings_post', 'addon/mathjax/mathjax.php', 'mathjax_settings_post');
+    Addon::registerHook('template_vars','addon/mathjax/mathjax.php', 'mathjax_template_vars');
     logger('installed js_math addon');
 }
+
+function mathjax_template_vars($a, &$arr) {
+    $arr['vars']['addon_hooks'][] = "mathjax";
```
Comment (MrPetovan) [inline mathjax/mathjax.php]: What if the `addon_hooks` key doesn't exist in the array? Wouldn't it trigger a Notice message?

File: mathjax/mathjax.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Addon_registerHook("postprocess","mathjax_postprocess_liveupdate");
```
Comment (MrPetovan) [inline mathjax/mathjax.js]: Code standards: Please add a space after commas.

File: mathjax/mathjax.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+function mathjax_postprocess_liveupdate()
+{
+	MathJax.Hub.Queue(["Typeset",MathJax.Hub]);
```
Comment (MrPetovan) [inline mathjax/mathjax.js]: Code standards: Please add a space after commas.

File: mathjax/mathjax.php [hunk 4]
```diff
@@ -16,12 +16,19 @@ function mathjax_install() {
     Addon::registerHook('page_header', 'addon/mathjax/mathjax.php', 'mathjax_page_header');
     Addon::registerHook('addon_settings', 'addon/mathjax/mathjax.php', 'mathjax_settings');
     Addon::registerHook('addon_settings_post', 'addon/mathjax/mathjax.php', 'mathjax_settings_post');
+    Addon::registerHook('template_vars','addon/mathjax/mathjax.php', 'mathjax_template_vars');
```
Comment (MrPetovan) [inline mathjax/mathjax.php]: Code standards: Please add a space after commas.

File: mathjax/mathjax.php [hunk 5]
```diff
@@ -16,12 +16,19 @@ function mathjax_install() {
...
+
+function mathjax_template_vars($a, &$arr) {
+    $arr['vars']['addon_hooks'][] = "mathjax";
+}
+
 function mathjax_uninstall() {
     Addon::unregisterHook('page_header', 'addon/mathjax/mathjax.php', 'mathjax_page_header');
     Addon::unregisterHook('addon_settings', 'addon/mathjax/mathjax.php', 'mathjax_settings');
     Addon::unregisterHook('addon_settings_post', 'addon/mathjax/mathjax.php', 'mathjax_settings_post');
+    Addon::unregisterHook('template_vars','addon/mathjax/mathjax.php', 'mathjax_template_vars');
```
Comment (MrPetovan) [inline mathjax/mathjax.php]: Code standards: Please add a space after commas.

File: mathjax/mathjax.php (hunk 1 above)
Comment (abanink) [reply mathjax/mathjax.php]: apparently it doesn't, at least in my environment (php 7.0)
I enabled `E_NOTICE` and had many NOTICE messages, but not in `addon/mathjax/mathjax.php`.

If you think I should add some defensive code anyway, I'll be happy to add it.

File: mathjax/mathjax.php [hunk 6]
```diff
@@ -16,13 +16,21 @@ function mathjax_install() {
...
+
 function mathjax_uninstall() {
     Addon::unregisterHook('page_header', 'addon/mathjax/mathjax.php', 'mathjax_page_header');
     Addon::unregisterHook('addon_settings', 'addon/mathjax/mathjax.php', 'mathjax_settings');
     Addon::unregisterHook('addon_settings_post', 'addon/mathjax/mathjax.php', 'mathjax_settings_post');
+    Addon::unregisterHook('template_vars', 'addon/mathjax/mathjax.php', 'mathjax_template_vars');
+}
+
+function mathjax_template_vars($a, &$arr) {
+    $arr['vars']['addon_hooks'][] = "mathjax";
```
Comment (MrPetovan) [inline mathjax/mathjax.php]: > If you think I should add some defensive code anyway, I'll be happy to add it.

Yeah, I'd rather, please ensure `$arr['vars']['addon_hooks']` is initialized.

**Pull request merged**